esp-hal-procmacros/src/lib.rs

```diff
@@ -381,3 +396,127 @@ pub fn main(args: TokenStream, item: TokenStream) -> TokenStream {
 
     run(&args.meta, f, main()).unwrap_or_else(|x| x).into()
 }
+
+/// Automatically implement the [Builder Lite] pattern for a struct.
+///
+/// This will create an `impl` which contains methods for each field of a
+/// struct, allowing users to easily set the values. The generated methods will
+/// be the field name prefixed with `with_`, and calls to these methods can be
+/// chained as needed.
+///
+/// ## Example
+///
+/// ```rust, no_run
+/// #[derive(Default)]
+/// enum MyEnum {
+///     #[default]
+///     A,
+///     B,
+/// }
+///
+/// #[derive(Default, BuilderLite)]
+/// #[non_exhaustive]
+/// struct MyStruct {
+///     enum_field: MyEnum,
+///     bool_field: bool,
+///     option_field: Option<i32>,
+/// }
+///
+/// MyStruct::default()
+///     .with_enum_field(MyEnum::B)
+///     .with_bool_field(true)
+///     .with_option_field(-5);
```

Comment:
Could we also have an example of `with_optional_field_none`? Just because it's the only thing that's different from the build lite blog post

Reply:
To properly demonstrate that we really should add another field and then implement `Default` explicitly for the struct, at which point this example becomes quite long which is in violation of our API guidelines. I would also be incredibly surprised if anybody ever looked at this comment.

Reply:
This macro also generates concrete impls on types and all defined methods are clearly documented.

<img width="1021" alt="Screenshot 2024-11-27 at 16 30 59" src="https://github.com/user-attachments/assets/4507cbbc-d4a6-4636-9dee-6df0ae27b857">

Reply:
Ah yeah, the docs look good enough, thanks!